src/cobalt.rs

```diff
@@ -131,9 +131,19 @@ pub fn build(config: &Config) -> Result<()> {
         .collect();
 
     trace!("Generating posts");
-    for mut post in &mut posts {
+    for (i, mut post) in &mut posts.iter_mut().enumerate() {
         trace!("Generating {}", post.path);
 
+        // posts are in reverse date order, so previous post is the next in the list (+1)
+        if let Some(previous) = simple_posts_data.get(i + 1) {
```

Comment:
Drat, I was hoping [Iter::window](https://doc.rust-lang.org/std/primitive.slice.html#method.windows) would solve this problem for us. Ideally, we'd use an iterator for this but this is fine for now.